File: app/src/main/java/com/hjc/base/adapter/CommonBindingAdapters.java
package com.hjc.base.adapter;

import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.bitmap.CircleCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;
import com.bumptech.glide.request.target.CustomTarget;
import com.bumptech.glide.request.transition.Transition;

import jp.wasabeef.glide.transformations.BlurTransformation;


/**
 * @Author: HJC
 * @Date: 2020/5/14 15:45
 * @Description: 自定义BindingAdapter
 */
public class CommonBindingAdapters {

    @BindingAdapter(value = {"imageUrl", "placeholder"}, requireAll = false)
    public static void loadImage(ImageView imageView, String url, Drawable drawable) {
        if (!TextUtils.isEmpty(url)) {
            RequestOptions requestOptions = new RequestOptions()
                    .placeholder(drawable)
                    .error(drawable)
                    .diskCacheStrategy(DiskCacheStrategy.RESOURCE);

            Glide.with(imageView.getContext())
                    .load(url)
                    .apply(requestOptions)
                    .transition(DrawableTransitionOptions.withCrossFade(500))
                    .into(imageView);
        }
    }

    @BindingAdapter("imageBgUrl")
    public static void loadBgImage(final ViewGroup viewGroup, String url) {
        if (!TextUtils.isEmpty(url)) {
            Glide.with(viewGroup.getContext())
                    .asBitmap()
                    .load(url)
                    .into(new CustomTarget<Bitmap>() {
                        @Override
                        public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? super Bitmap> transition) {
                            Drawable drawable = new BitmapDrawable(viewGroup.getResources(), resource);
                            viewGroup.setBackground(drawable);
                        }

                        @Override
                        public void onLoadCleared(@Nullable Drawable placeholder) {

                        }
                    });
        }
    }

    @BindingAdapter(value = {"imageRoundUrl", "placeholder", "roundingRadius"}, requireAll = false)
    public static void loadRoundImage(ImageView imageView, String url, Drawable drawable, int roundingRadius) {
        if (!TextUtils.isEmpty(url)) {
            RequestOptions requestOptions = new RequestOptions()
                    .placeholder(drawable)
                    .error(drawable)
                    .transform(new RoundedCorners(roundingRadius))
                    .diskCacheStrategy(DiskCacheStrategy.RESOURCE);

            Glide.with(imageView.getContext())
                    .load(url)
                    .placeholder(drawable)
                    .apply(requestOptions)
                    .into(imageView);
        }
    }

    @BindingAdapter(value = {"imageCircleUrl", "placeholder"}, requireAll = false)
    public static void loadCircleImage(ImageView imageView, String url, Drawable drawable) {
        if (!TextUtils.isEmpty(url)) {
            RequestOptions requestOptions = new RequestOptions()
                    .placeholder(drawable)
                    .error(drawable)
                    .transform(new CircleCrop())
                    .diskCacheStrategy(DiskCacheStrategy.RESOURCE);

            Glide.with(imageView.getContext())
                    .load(url)
                    .apply(requestOptions)
                    .into(imageView);
        }
    }


    @BindingAdapter(value = {"imageBlurUrl", "placeholder", "radius", "sampling"}, requireAll = false)
    public static void loadBlurImage(ImageView imageView, String url, Drawable drawable, int radius, int sampling) {
        RequestOptions requestOptions = new RequestOptions()
                .placeholder(drawable)
                .error(drawable)
                .transform(new BlurTransformation(radius, sampling))
                .diskCacheStrategy(DiskCacheStrategy.RESOURCE);

        Glide.with(imageView.getContext())
                .load(url)
                .apply(requestOptions)
                .transition(DrawableTransitionOptions.withCrossFade(500))
                .into(imageView);
    }

    @BindingAdapter("android:visibility")
    public static void setVisibility(View view, boolean value) {
        view.setVisibility(value ? View.VISIBLE : View.GONE);
    }
}

File: library-common/src/main/java/com/hjc/library_common/adapter/CommonBindingAdapters.kt
package com.hjc.library_common.adapter

import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.Drawable
import android.text.TextUtils
import android.view.ViewGroup
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.CircleCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.CustomTarget
import com.bumptech.glide.request.transition.Transition
import jp.wasabeef.glide.transformations.BlurTransformation

/**
 * @Author: HJC
 * @Date: 2021/1/8 16:23
 * @Description: 自定义BindingAdapter(布局里直接使用)
 */
object CommonBindingAdapters {

    /**
     * 加载图片
     */
    @JvmStatic
    @BindingAdapter(value = ["imageUrl", "placeholder"], requireAll = false)
    fun loadImage(imageView: ImageView, url: String?, drawable: Drawable?) {
        if (!TextUtils.isEmpty(url)) {
            val requestOptions = RequestOptions()
                .placeholder(drawable)
                .error(drawable)
                .diskCacheStrategy(DiskCacheStrategy.RESOURCE)
            Glide.with(imageView.context)
                .load(url)
                .apply(requestOptions)
                .transition(DrawableTransitionOptions.withCrossFade(500))
                .into(imageView)
        }
    }

    /**
     * 加载背景图片
     */
    @JvmStatic
    @BindingAdapter("imageBgUrl")
    fun loadBgImage(viewGroup: ViewGroup, url: String?) {
        if (!TextUtils.isEmpty(url)) {
            Glide.with(viewGroup.context)
                .asBitmap()
                .load(url)
                .into(object : CustomTarget<Bitmap?>() {

                    override fun onLoadCleared(placeholder: Drawable?) {}

                    override fun onResourceReady(
                        resource: Bitmap,
                        transition: Transition<in Bitmap?>?
                    ) {
                        val drawable: Drawable = BitmapDrawable(viewGroup.resources, resource)
                        viewGroup.background = drawable
                    }
                })
        }
    }

    /**
     * 加载圆角图片
     */
    @JvmStatic
    @BindingAdapter(value = ["imageRoundUrl", "placeholder", "roundingRadius"], requireAll = false)
    fun loadRoundImage(
        imageView: ImageView,
        url: String?,
        drawable: Drawable?,
        roundingRadius: Int
    ) {
        if (!TextUtils.isEmpty(url)) {
            val requestOptions = RequestOptions()
                .placeholder(drawable)
                .error(drawable)
                .transform(CenterCrop(), RoundedCorners(roundingRadius))
                .diskCacheStrategy(DiskCacheStrategy.RESOURCE)
            Glide.with(imageView.context)
                .load(url)
                .placeholder(drawable)
                .apply(requestOptions)
                .into(imageView)
        }
    }

    /**
     * 加载圆形图片
     */
    @JvmStatic
    @BindingAdapter(value = ["imageCircleUrl", "placeholder"], requireAll = false)
    fun loadCircleImage(imageView: ImageView, url: String?, drawable: Drawable?) {
        if (!TextUtils.isEmpty(url)) {
            val requestOptions = RequestOptions()
                .placeholder(drawable)
                .error(drawable)
                .transform(CircleCrop())
                .diskCacheStrategy(DiskCacheStrategy.RESOURCE)
            Glide.with(imageView.context)
                .load(url)
                .apply(requestOptions)
                .into(imageView)
        }
    }

    /**
     * 加载高斯模糊图片
     */
    @JvmStatic
    @BindingAdapter(
        value = ["imageBlurUrl", "placeholder", "radius", "sampling"],
        requireAll = false
    )
    fun loadBlurImage(
        imageView: ImageView,
        url: String?,
        drawable: Drawable?,
        radius: Int,
        sampling: Int
    ) {
        val requestOptions = RequestOptions()
            .placeholder(drawable)
            .error(drawable)
            .transform(BlurTransformation(radius, sampling))
            .diskCacheStrategy(DiskCacheStrategy.RESOURCE)
        Glide.with(imageView.context)
            .load(url)
            .apply(requestOptions)
            .transition(DrawableTransitionOptions.withCrossFade(500))
            .into(imageView)
    }

}